src/main/java/com/garrytrue/repository/StudentRepository.java

```diff
@@ -0,0 +1,40 @@
+package com.garrytrue.repository;
+
+
+import com.garrytrue.model.Student;
+
+import java.util.Map;
+
+/**
+ * Created by garrytrue on 12.03.17.
+ */
+public class StudentRepository implements CRUDRepository<Student> {
```

Comment:
you need atomic long to use as autoincrement of Student.id filed.
this is a state of your repository.

User creates Student - no id is present; only after you save it to repository. repository increments current ID and assigns it to new Student.